controllers/awsmachine_controller.go

```diff
@@ -1231,6 +1231,10 @@ func (r *AWSMachineReconciler) ensureInstanceMetadataOptions(ec2svc services.EC2
 func (r *AWSMachineReconciler) getControlPlane(ctx context.Context, log *logger.Logger, cluster *clusterv1.Cluster) (*unstructured.Unstructured, error) {
 	var ns string
 
+	if cluster.Spec.ControlPlaneRef == nil {
```

Comment:
Wouldn't be better returning an error?

A cluster with no Control Plane definition should be incorrect.

Reply:
> A cluster with no Control Plane definition should be incorrect.

Even for a self-managed cluster?

Reply:
Do you mean with the cluster API managed label?

Reply:
Yes, I think it makes more sense to have an error return although im not sure how someone would get into this situation in the first place

Reply:
Same here, @patrickdillon, may I ask you to share why you should have a `Cluster` object with no `ControlPlane` reference?

Reply:
To clarify here, a cluster without a control plane reference is valid by contract, the reference is optional and it was made by design

Reply:
I think this is handled over at https://github.com/kubernetes-sigs/cluster-api-provider-aws/pull/4822/files#diff-8428e7154b9d78ae8ea33a3534b5051bd14a8b282d4d9f27d01889a8d1e3b2c1R139-R142